src/Proto.Actor/EventStream/DeadLetter.cs

```diff
@@ -46,7 +47,12 @@ public DeadLetterProcess(ActorSystem system) : base(system)
         protected internal override void SendUserMessage(PID pid, object message)
         {
             if (System.Metrics.Enabled)
-                ActorMetrics.DeadletterCount.Add(1, new("id", System.Id), new("address", System.Address), new("messagetype", message.GetType().Name));
+            {
+                ActorMetrics.DeadletterCount.Add(1,
+                    new("id", System.Id), new("address", System.Address),
+                    new("messagetype", MessageEnvelope.UnwrapMessage(message)?.GetType().Name ?? "<null>")
```

Comment:
Put it after `var (msg, sender, header) = MessageEnvelope.Unwrap(message);`, no need to unwrap it twice :)

Reply:
fixed